docs/specs/mlnet-cli/MLNET-CLI-Specs.md

```diff
@@ -0,0 +1,969 @@
+--> Cesar: Let's use column.
+
+- For the debug-log-file option, what's the file extenssion (.log ?) and internal format of the file? - Confirm this.
+
+- Will overriden default values just be provided as static values in the CLI commands or also based on a related configuration .JSON file placed along with the CLI executable? See [comparable .JSON files for dotnet CLI templates](https://github.com/dotnet/dotnet-template-samples/blob/master/05-multi-project/.template.config/template.json))
+
+- Support Custom templates for "automlnet new" such as [Custom templates for dotnet new](https://docs.microsoft.com/en-us/dotnet/core/tools/custom-templates)? - That could allow extensibility for other application project types or even for other languages like F# or additional scenarios.
+CESAR --> No, the CLI team confirmed that custom templates were meant to be created by static code, not for code generation. It would be too complex.
+
+
+
+
+## Additional characteristics to be defined
+
+- Stopping criteria  - { Default timeout or timeout provided by the user }
+
+Gleb's (Cesar: Although, ins't this related to AutoML API instead the CLI?):
+We should have the following (in priority order):
+
+0. Azure integration (vNext)
+
+1. Time
+2. Number of iterations/tries - this is useful to ensure we have at least one iterations. It is also data size invariant.
+
+    And at least one from the list below. 6. and 7 are really good if you want to get the best possible accuracy.
+
+3. Tolerant - Stops when "validationScore" < "bestScore" - TH
+4. Loss of Generality - Stops when "validationScore" < "bestScore" * (1 - TH)
+5. Low Progress - Stops when "recentAvg" >= (1 -TH) * "recentBest"
+6. Generality to Progress rate - Stops when "validationScore" * "recentBest" <= (1 - TH) *
+7. Consecutive Loss in Generality - Stops when the score on validation set degraded W times in a row.
+
+- Stopping criteria for single iteration { timeout, low accuracy }
+- Concurrency – { Max concurrent iterations, max cores per iteration)
+- Cross validation { # folds, validation size %, stratification column }
+- Trainers inclusion list and trainers exclusion list
+- Model explainability { true/false }
+- Ensembling { true/false, iterations }
+
+
+# References
+
+- [POSIX Utility Conventions](http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html)
+- [dotnet new](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-new?tabs=netcore21)
+- [Microsoft AutoML for Python](https://docs.microsoft.com/en-us/azure/machine-learning/service/concept-automated-ml)
+- [Uber Ludwig CLI Blog Post](https://eng.uber.com/introducing-ludwig/)
+- [Uber Ludwig CLI Getting Started](https://uber.github.io/ludwig/getting_started/)
+- [Uber Ludwig CLI syntax](https://uber.github.io/ludwig/user_guide/)
+
```

Comment:
Adding this as a potential reference:

TransmogrifAI (pronounced trăns-mŏgˈrə-fī) is an AutoML library for building modular, reusable, strongly typed machine learning workflows on Spark with minimal hand tuning

[TransmogrifAI Site](https://transmogrif.ai/)
[TransmogrifAI Repo](https://github.com/salesforce/TransmogrifAI)